### PR TITLE
Fixes #11852 - start extensions explicitly and docs

### PR DESCRIPTION
**Pull request by lzap**
Makes sure extensions gets started even without fetch-extension unit (when
network is down). Also added docs about how to create them (finally!).
@timogoebel

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,6 +171,20 @@ And finally (do not forget):
 
 		LC_ALL=C repoowner=theforeman branch=master proxy_repo=1.14 vagrant destroy $distro
 
+Extensions
+----------
+
+Discovery Image supports runtime extensions published via TFTP or HTTP.
+Those are distributes as ZIP file with shell scripts. It is also possible
```
Comment (timogoebel) [inline README.md]: typo: distribute_d_ ad ZIP file_s_

File: README.md [hunk 2]
```diff
@@ -171,6 +171,20 @@ And finally (do not forget):
...
+to build an image with extension built-in which is helpful for PXE-less
+environments.
+
+To do that, (follow the documentation)[https://theforeman.org/plugins/foreman_discovery/8.0/index.html#5.Extendingtheimage]
+to create directory structure in root/opt/extension folder. Do not put ZIP
+files into this folder, but keep the directory structure extracted (this is
+the directory where ZIP files gets downloaded and extracted). Then rebuild
```
Comment (timogoebel) [inline README.md]: typo: get__ downloaded

**Comment by lzap** [comment]
@timogoebel does the image build for you? Something in CentOS 7.3 grew in size, I need to increase the volume (1 GB) in `10-header.ks` a bit in order to build it...

**Comment by lzap** [comment]
I don't see any deviations, the list in `/usr/PACKAGES-LIST` basically grew in size, mainly kernel. If you see something that can be removed let me know.

**Comment by lzap** [comment]
Fixed typos, increased default root volume size to 1200 MB. Kernel is growing, we need to start thinking about dropping some drivers from it (unrelated drivers perhaps).

**Comment by timogoebel** [comment]
@lzap : Sorry, I didn't get to building the image, yet and don't have time to to it this week.

File: README.md [hunk 3]
```diff
@@ -171,6 +171,20 @@ And finally (do not forget):
 
 		LC_ALL=C repoowner=theforeman branch=master proxy_repo=1.14 vagrant destroy $distro
 
+Extensions
+----------
+
+Discovery Image supports runtime extensions published via TFTP or HTTP.
+Those are distributed as ZIP file with shell scripts. It is also possible
```
Comment (timogoebel) [inline README.md]: Those are distributed as ZIP files with shell scripts. It is also possible to build an image with extensions built-in which is helpful for PXE-less environments.

File: README.md [hunk 4]
```diff
@@ -171,6 +171,20 @@ And finally (do not forget):
...
+to build an image with extension built-in which is helpful for PXE-less
+environments.
+
+To do that, (follow the documentation)[https://theforeman.org/plugins/foreman_discovery/8.0/index.html#5.Extendingtheimage]
```
Comment (timogoebel) [inline README.md]: To do that, create a directory structure inside of `root/opt/extension` or follow the [complete documentation](https://theforeman.org/plugins/foreman_discovery/8.0/index.html#5.Extendingtheimage). 

Text and link are off text needs `[]` - url needs `()`.

**Comment by timogoebel** [comment]
@lzap: Image builds nicely. The docs need some minor improvements and then this is ready to merge.

**Comment by lzap** [comment]
Fixed docs, many thanks for testing this is *huge* help.

**Pull request merged**